plugins/org.jkiss.dbeaver.ext.cubrid/src/org/jkiss/dbeaver/ext/cubrid/model/meta/CubridMetaModel.java

```diff
@@ -149,4 +205,226 @@ public GenericTableBase createTableOrViewImpl(
+            throws DBException
+    {
+        String name = JDBCUtils.safeGetStringTrimmed(dbResult, "synonym_name");
+        String description = JDBCUtils.safeGetString(dbResult, "comment");
+        return new CubridSynonym(container, name, description, dbResult);
+    }
+
+    @Override
+    public boolean supportsDatabaseTriggers(@NotNull GenericDataSource dataSource)
+    {
+        return true;
+    }
+
+    @Override
+    public boolean supportsTriggers(@NotNull GenericDataSource dataSource)
+    {
+        return true;
+    }
+
+    @Override
+    public JDBCStatement prepareTableTriggersLoadStatement(
+            @NotNull JDBCSession session,
+            @NotNull GenericStructContainer container,
+            @Nullable GenericTableBase table)
+            throws SQLException
+    {
+        String sql = "select t1.*, t2.*, owner.name from db_trigger t1 join db_trig t2 \n"
+                + "on t1.name = t2.trigger_name where owner.name = ? \n"
+                + (table != null ? "and target_class_name = ?" : "");
+        final JDBCPreparedStatement dbStat = session.prepareStatement(sql);
+        dbStat.setString(1, container.getName());
+        if (table != null) {
+            dbStat.setString(2, table.getName());
+        }
+        return dbStat;
+    }
+
+    @Override
+    public CubridTrigger createTableTriggerImpl(
+            @NotNull JDBCSession session,
+            @NotNull GenericStructContainer container,
+            @NotNull GenericTableBase table,
+            String triggerName,
+            @NotNull JDBCResultSet dbResult)
+            throws DBException
+    {
+        String name = JDBCUtils.safeGetString(dbResult, "name");
+        String description = JDBCUtils.safeGetString(dbResult, "comment");
+        return new CubridTrigger(table, name, description, dbResult);
+    }
+
+    @Override
+    public JDBCStatement prepareContainerTriggersLoadStatement(
+            @NotNull JDBCSession session,
+            @Nullable GenericStructContainer container)
+            throws SQLException
+    {
+        String sql = "select t1.*, t2.*, owner.name from db_trigger t1 join db_trig t2 \n"
+                + "on t1.name = t2.trigger_name where owner.name = ? \n";
+        final JDBCPreparedStatement dbStat = session.prepareStatement(sql);
+        dbStat.setString(1, container.getName());
+        return dbStat;
+    }
+
+    @Override
+    public CubridTrigger createContainerTriggerImpl(
+            @NotNull GenericStructContainer container,
+            @NotNull JDBCResultSet dbResult)
+            throws DBException
+    {
+        String name = JDBCUtils.safeGetString(dbResult, "name");
+        String description = JDBCUtils.safeGetString(dbResult, "comment");
+        String tableName = JDBCUtils.safeGetString(dbResult, "target_class_name");
+        DBRProgressMonitor monitor = dbResult.getSession().getProgressMonitor();
+        CubridTable cubridTable = null;

```

Comment:
DBAs and DBA members can create triggers in different schemas. Therefore, cubridTable can be null.
I think I need to get all the table information.

Reply:
Noted, I will check this.